app/controllers/api/v1/documents.rb

```diff
@@ -232,8 +232,20 @@ def document_repository
 
         put ':document_id', requirements: { document_id: /.*/ } do
           id = params.delete(:document_id)
-          # return just enough attributes to ensure the document is valid
-          document = document_repository.find(id, _source: %w[language path created_at])
+          # SRCH-5096 Ensure that existing attributes are not overwritten on put or else the weekly
+          # searchgov ClickMonitorJob and (infrequent) `searchgov:promote` task will delete metadata
```

Comment:
I don’t know much about i14y, so can you explain how including more parameters in the `find` stops the attributes from being overwritten? PR LGTM otherwise.

Reply:
@ddzz Good question (with full transparency that I have no clue why it was written like this in the first place so I'm trying to keep my changes as minimal as possible while still fixing this issue).

Without these attributes in here when the document variable is set it only will include what is returned by `find`.  So, even if an existing i14y document is loaded up with every type of filter field, date field, etc. that it is possible to have, we'll only have a document that includes `language`, `path`, and `created_at`.  To that minimal skeleton of a document is added whatever specific parameters are passed along in the PUT (say, adding `click_count` from the ClickMonitorJob or setting `promote` to true via the promote rake task).  That document - now containing only `language`, `path`, `created_at`, and the update parameters - is then added to the document repository.  Any preexisting values, say for `audience` or `searchgov_custom1` are dropped altogether because they were not part of the document that was updated.